particle_life.html

```diff
@@ -1,60 +1,126 @@
-  <canvas id="life" width="500" height="500"></canvas>
-  <script>
-    //Hunar Ahmad @ brainxyz
-  m=document.getElementById("life").getContext('2d')
-  draw=(x,y,c,s)=>{m.fillStyle=c; m.fillRect(x, y, s, s)}
-  atoms=[]
-  atom=(x,y,c)=>{return{"x":x, "y":y, "vx":0, "vy":0, "color":c}}
-  random=()=>{return Math.random()*400+50}
-  create=(number, color)=>{ 
-  	group=[]
-    for(let i=0; i < number; i++){
-	    group.push(atom(random(), random(), color))
-	    atoms.push(group[i]) 
-	}
-    return group 
-  }
-  rule=(atoms1, atoms2, g)=>{
-  for(let i=0; i < atoms1.length; i++){
-    fx = 0
-    fy = 0
-    for(let j=0; j < atoms2.length; j++){
-      a = atoms1[i]
-      b = atoms2[j]
-      dx = a.x-b.x
-      dy = a.y-b.y
-      d = Math.sqrt(dx*dx + dy*dy)
-      if(d > 0 && d < 80){
-      	F = g * 1/d
-        fx += (F * dx)
-        fy += (F * dy)
+<canvas id="life" width="700" height="700"></canvas>
+<script>
+  //Hunar Ahmad @ brainxyz
+
+  const RULES = {
+    green: {
+      green: 0.878214014158254,
+      red: 0.383942932294564,
+      yellow: 0.3632328353781209,
+      blue: 0.4357079645785089,
+    },
+    red: {
+      green: -0.8131279812066854,
+      red: 0.8761564046567396,
+      yellow: -0.686246916739194,
+      blue: -0.42403398294928163,
+    },
+    yellow: {
+      green: 0.8283611643992606,
+      red: -0.8050409003234531,
+      yellow: 0.8422661062679588,
+      blue: -0.6206303204367405,
+    },
+    blue: {
+      green: -0.6276679142294777,
+      red: -0.48726835984229977,
+      yellow: -0.8155039608681607,
+      blue: 0.49503848830455155,
+    },
+  };
+
+  // const RULES = {
+  //   green: {
+  //     green: Math.random() * 2 - 1,
+  //     red: Math.random() * 2 - 1,
+  //     yellow: Math.random() * 2 - 1,
+  //     blue: Math.random() * 2 - 1,
+  //   },
+  //   red: {
+  //     green: Math.random() * 2 - 1,
+  //     red: Math.random() * 2 - 1,
+  //     yellow: Math.random() * 2 - 1,
+  //     blue: Math.random() * 2 - 1,
+  //   },
+  //   yellow: {
+  //     green: Math.random() * 2 - 1,
+  //     red: Math.random() * 2 - 1,
+  //     yellow: Math.random() * 2 - 1,
+  //     blue: Math.random() * 2 - 1,
+  //   },
+  //   blue: {
+  //     green: Math.random() * 2 - 1,
+  //     red: Math.random() * 2 - 1,
+  //     yellow: Math.random() * 2 - 1,
+  //     blue: Math.random() * 2 - 1,
+  //   },
+  // };
+
+  console.log(JSON.stringify(RULES));

```

Comment:
If you uncomment the above block with the random rules, you can reload the page until you see an interesting pattern, and then copy the RULES from the dev tools console.

Reply:
Thanks a lot for the contribution. The code indeed is faster. 
It will be interesting if we could add some realtime GUI controls to the JavaScript as well because in my experience most of the interesting patterns in the C++ were found because of the realtime visual feedback from changing the controls and seeing what happens.

Reply:
Good idea! I'll open another PR for the controls.